webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ComplianceApi.scala

```diff
@@ -679,6 +750,179 @@ class ComplianceAPIService(
     }
   }
 
+  private[this] def getByNodeGroupCompliance(
+      nodeGroup:          NodeGroup,
+      level:              Option[Int],
+      isGlobalCompliance: Boolean
+  ): IOResult[ByNodeGroupCompliance] = {
+    for {
+      t1        <- currentTimeMillis
+      allGroups <- nodeGroupRepo.getAllNodeIdsChunk()
+      t2        <- currentTimeMillis
+      _         <- TimingDebugLoggerPure.trace(s"getByNodeGroupCompliance - nodeGroupRepo.getAllNodeIds in ${t2 - t1} ms")
+
+      directiveLib <- directiveRepo.getFullDirectiveLibrary()
+      t3           <- currentTimeMillis
+      _            <- TimingDebugLoggerPure.trace(s"getByNodeGroupCompliance - getFullDirectiveLibrary in ${t3 - t2} ms")
+
+      nodeInfos <- nodeInfoService.getAll()
+      t4        <- currentTimeMillis
+      _         <- TimingDebugLoggerPure.trace(s"getByNodeGroupCompliance - nodeInfoService.getAll() in ${t4 - t3} ms")
+
+      compliance <- getGlobalComplianceMode().toIO
+      t5         <- currentTimeMillis
+      _          <- TimingDebugLoggerPure.trace(s"getByNodeGroupCompliance - getGlobalComplianceMode in ${t5 - t4} ms")
+
+      rules <- rulesRepo.getAll()
+      t6    <- currentTimeMillis
+      _     <- TimingDebugLoggerPure.trace(s"getByNodeGroupCompliance - getAllRules in ${t6 - t5} ms")
+
+      // - global compliance : does the group contain a node that any rule applied to ? yes: include the rule in the compliance, and only nodes that belong to group
+      // - targeted compliance : does the rule target the group directly (for group targets : includes it, and not in exclude) ? yes: same
+
+      // filter our rules that are applicable to any node in this group and get their compliances
+      currentGroupNodeIds = allGroups.getOrElse(nodeGroup.id, Chunk.empty)
+
+      targetedNodeIdsByRuleId =
+        rules.map(rule => (rule.id, RoNodeGroupRepository.getNodeIdsChunk(allGroups, rule.targets, nodeInfos))).toMap
+
+      filteredRules = rules.filter(rule => {
+                        val targetedNodeIds      = targetedNodeIdsByRuleId(rule.id)
+                        val isRuleTargetingGroup =
+                          RuleTarget.merge(rule.targets).includes(GroupTarget(nodeGroup.id)) // targeted compliance
+                        (isGlobalCompliance || isRuleTargetingGroup) && targetedNodeIds.exists(currentGroupNodeIds.contains(_))
```

Comment:
(:exploding_head:)

This adds the check for `isRuleTargetingGroup` only when compliance is not global (i.e. targeted compliance)

There are multiple of occurences of : 
```
(condA || condB) && condC 
``` 
in this PR and the way to read them is : "I always want to check `condC`", and "I only want to check `condB` when `condA`"